hibernate5/src/test/java/com/baeldung/hibernate/proxy/HibernateProxyUnitTest.java

```diff
@@ -0,0 +1,75 @@
+package com.baeldung.hibernate.proxy;
+
+import org.hibernate.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
+public class HibernateProxyUnitTest {
+
+    private Session session;
+
+    @Before
+    public void init(){
+        try {
+            session = HibernateUtil.getSessionFactory("hibernate.properties")
+                    .openSession();
+        } catch (HibernateException | IOException e) {
+            fail("Failed to initiate Hibernate Session [Exception:" + e.toString() + "]");
+        }
+
+        Boss boss = new Boss("Eduard", "Freud");
+        session.save(boss);
+    }
+
+    @After
+    public void close(){
+        if(session != null) {
+            session.close();
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void givenAnInexistentEmployeeId_whenUseGetMethod_thenReturnNull() {
+        Employee employee = session.get(Employee.class, new Long(14));
+        assertNull(employee);
+        employee.getId();
+    }
+
+    @Test
+    public void givenAnInexistentEmployeeId_whenUseLoadMethod_thenReturnAProxy() {
+        Employee employee = session.load(Employee.class, new Long(14));
+        assertNotNull(employee);
+    }
+
```

Comment:
I think we are missing a couple of tests here.

In the tutorial, there is a statement that when using load and then invoking a method, we get an exception. That would be an informative test.

More importantly, though, the examples regarding what happens with relationships, with batch size, etc. aren't represented in the tests. We should add those so the reader can see those in action.